packages/live-preview/src/subscribe.ts

```diff
@@ -10,8 +10,15 @@ export const subscribe = <T>(args: {
   const { apiRoute, callback, depth, initialData, serverURL } = args
 
   const onMessage = async (event: MessageEvent) => {
-    const mergedData = await handleMessage<T>({ apiRoute, depth, event, initialData, serverURL })
-    callback(mergedData)
+    if (
+      event.origin === serverURL &&
+      event.data &&
```

Comment:
`event.data` is redundant with the next condition `typeof event.data === 'object'`.

Reply:
True, copy & pasted...